src/routers/alpha-router/alpha-router.ts

```diff
@@ -1356,6 +1361,15 @@ export class AlphaRouter
       );
     }
 
+    if (
```

Comment:
should we not even fetch them in this case instead of fetching and then setting to null?

Reply:
also does that mean that we won't use cached routes at all in this case?
would that impact latency?

Reply:
we should still fetch cached routes for mixed protocol request. the reason is because there's a chance that cached routes do not contain any excluded protocol-versioned pool in any of its cached routes. We should only nullify the cached routes, when any of the cached routes is a mixed route, and such mixed route contains a protocol-versioned pool, that should be excluded.

Although I did realize then, that we should only nullify the cached routes, only when the intent=caching, but not when intent=quote. intent=quote means it's an online request from routing-api quote endpoint perspective. If it's a cache hit, we still want routing-api to return the ineligible cached route, if it contains a mixed route, such that the mixed route contains a protocol-versioned pool that should be excluded. This is during intent=quote. For every intent=quote, routing-api self invokes its own lambda again via intent=caching. This is when we should nullify the cached routes, and re-compute the routes, if any of any of the cached routes is a mixed route, and such mixed route contains a protocol-versioned pool, that should be excluded.

This seems like a good tradeoff between online quote endpoint perf vs the correctness returned from the quote endpoint for the first quote request. In reality, because the top route (that includes the mixed route) is so low as compared to a pure v2 or v3 route, I think this tradeoff of favoring endpoint perf over first quote response correctness is warranted:

<img width="1434" alt="Screenshot 2024-09-05 at 4 28 58 PM" src="https://github.com/user-attachments/assets/93056143-2c0b-4650-b675-6e4ea4e22797">

I will make a change to only nullify the cached routes during intent=caching

Reply:
done